packages/node/src/lib/http-client.ts

```diff
@@ -49,10 +51,9 @@ export interface HTTPClientRequest {
    */
   headers: Record<string, string>
   /**
-   * JSON data to be sent with the request (will be stringified)
-   * @example { "batch": [ ... ]}
+   * Data to be sent with the request
    */
-  data: Record<string, any>
+  body: string
```

Comment:
Ah, unfortunately this is a breaking change. If someone is using the custom HTTP client API, then they'd be passing the `data` property to their HTTP client now.

I do think this should have been a string to start with. Would be open to making a major version bump to update this, but we would need to be clear in the changelog this only impacts customers using the custom HTTP client, and I think show a snippet for migrating purposes of what the code looks like before and after.

The other alternative I see is having a separate HTTP client for just the oauth, but I think that would be bad devex to have 2 slightly different HTTP client interfaces (especially since we're only aware of using them to support proxies at this point)

Note: If we do go the major version bump route, would also be tempted to open up `method` to a string in case we end up having to support other requests in the future.

Reply:
The driving factor here was that the body of the request to the token endpoint currently must be URLencoded.

Reply:
@chrisradek agree it's a breaking change, but FWIW very likely that nobody is using this API  -- even in the docs there's a disclaimer that says "you don't need this".  I'm fine going up a major version, though.

Reply:
To be clear, I'm totally fine doing a major version bump for this, just calling it out since it's currently marked as a minor bump in the changeset. No need to beat ourselves up over it :smile: Would just like to limit doing them too often if we can avoid it.

With regards to the other methods, I did learn this really neat trick the other day - you can write a type like this:
`type methods = 'POST' | string & {}` and you'll get completion in your editor, but also still be able to pass any string 
(Try it here! https://www.typescriptlang.org/play?ssl=1&ssc=36&pln=1&pc=1#code/C4TwDgpgBAthwAsD2ATAzlAvFA5ABQHkBlAFRygB8o1gAnASwDsBzKAMigG8BfAKF4BmAV0YBjYPSSMowCDQAUcRKgBcseMnQBKLn16yF+YmS1A)

Though read the TypeScript team is working on making `type methods = 'POST' | string` work the same way. 🤞